lib/OpenQA/Worker/Jobs.pm

```diff
@@ -164,6 +175,21 @@ sub upload {
 
     my $res = $OpenQA::Worker::Common::ua->start($tx);
 
+    # Upload known server failures (Instead of anything that's not 200)
+    unless ($res->res->code =~ m/^5/) {
+        if ($retry_counter++ == $retry_limit) {
+            my $msg = sprintf("Uploads have failed for %s\n", $filename);
+            open(my $log, '>>', "autoinst-log.txt");
+            print $log $msg;
+            close $log;
+            print STDERR $msg;
+            return 0;
+        }
+        else {
+            goto START_UPLOAD;
```

Comment:
I do agree with @kraih about the goto (i personally don't like them that much) but i see that having a loop for something that is seldomly happening is not exactly good either (after all, its about treating failures as retries).

@kraih thoughts?